src/pe/delay_import.rs

```diff
@@ -0,0 +1,529 @@
+        if descriptor.is_null() || !descriptor.is_possibly_valid() {
+            self.bytes = &[];
+            return None;
+        }
+
+        let name_table_rva = descriptor.name_table_rva;
+        let name_table_offset = utils::find_offset(
+            name_table_rva as usize,
+            self.ctx.sections,
+            self.ctx.file_alignment,
+            &self.ctx.opts,
+        )?;
+        if self.ctx.bytes.len() < name_table_offset {
+            self.bytes = &[];
+            return None;
+        }
+
+        let res = DelayImportDll {
+            descriptor,
+            ctx: self.ctx,
+            bytes: &self.ctx.bytes[name_table_offset..],
+        };
+
+        Some(Ok(res))
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct DelayImportFunctionIterator<'a> {
+    ctx: PeCtx<'a>,
+    bytes: &'a [u8],
+    offset: usize,
+    descriptor: DelayImportDescriptor,
+}
+
+impl<'a> Iterator for DelayImportFunctionIterator<'a> {
+    type Item = error::Result<DelayImportFunction<'a>>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.offset >= self.bytes.len() {
+            return None;
+        }
+
+        let ctx = Ctx::new(self.ctx.container, scroll::LE);
+        let thunk = self
+            .bytes
+            .gread_with::<DelayImportThunk>(&mut self.offset, ctx)
+            .ok()?;
+        if thunk.is_null() {
+            self.bytes = &[];
+            return None;
+        }
+
+        let ctx = (self.descriptor, self.ctx);
+        let func = self.bytes.pread_with::<DelayImportFunction>(0, ctx).ok()?;
+
+        Some(Ok(func))
+    }
+}
+
+/// Represents a single entry in the delay import table of a PE (Portable Executable) file.
+#[derive(Debug)]
+pub struct DelayImportEntry<'a> {
+    /// The associated delay import descriptor.
+    pub descriptor: DelayImportDescriptor,
+    /// Name offset of this delay import entry.
+    pub offset: u32,
+    /// RVA of this delay import entry.
+    pub rva: u32,
+    /// Hint value for symbol lookup.
+    pub hint: u16,
+    /// Name of the DLL this function is imported from.
+    pub dll: &'a str,
+    /// Name of the imported function, if available.
+    ///
+    /// This is `None` if the import is by ordinal.
+    pub name: Option<&'a str>,
+    /// Ordinal number if the function is imported by ordinal.
+    ///
+    /// This is `0` if `name` is `Some` and vice versa.
+    pub ordinal: u16,
+}
+
+/// Internal helper struct to simplify bitflag operations.
+#[derive(Debug, SizeWith)]
+struct DelayImportThunk(pub u64);
+
+impl<'a> ctx::TryFromCtx<'a, Ctx> for DelayImportThunk {
+    type Error = crate::error::Error;
+    fn try_from_ctx(bytes: &'a [u8], ctx: Ctx) -> error::Result<(Self, usize)> {
+        let mut offset = 0;
+
+        let value = if ctx.is_big() {
+            bytes.gread_with::<u64>(&mut offset, scroll::LE)?
+        } else {
+            let v = bytes.gread_with::<u32>(&mut offset, scroll::LE)?;
+            v as u64
+        };
```

Comment:
not your concern, but i've seen this pattern enough I feel like this would be a good use for an annotation in scroll, e.g.:

```rust
#[derive(Debug, SizeWith, Pread)]
struct DelayImportThunk(#[scroll(upcast)] pub u64);
```
or perhaps

```rust
#[scroll(from("function"))]
```